docs/source/processing_at_scale/processing_with_aws.rst

```diff
@@ -0,0 +1,351 @@
+      "type": "container",
+      "containerProperties": {
+        "jobRoleArn": "arn:aws:iam::422553907334:role/spacetx-batch",
+        "image": "spacetx/process-fov",
+        "vcpus": 1,
+        "memory": 2500
+      }
+    }
+
+NOTE: if your starfish processing is memory expensive you can adjust the allocated memory for each created instance using the `memory` parameter.
+
+Then from the directory where this file lives run the following command:
+
+``aws batch submit-job --cli-input-json file://register-process-fov-job.json``
+
+You can check that your jobs had been successfully registered by navigating to the `Job Definitions page <https://console.aws.amazon.com/batch/home?region=us-east-1#/job-definitions>`_.
+
+Here's what our's looks like:
+
+.. figure:: images/aws-tutorial-figure3.png
+   :align: center
+
+Now open the file named `register-merge-job.json`. This file describes a batch job that will create an ec2 instance using the docker container `spacetx/merge-batch-job` that merges together all your processed results into
+one `DecodedIntensityTable`. Again replace the string "ADD ARN" with the aws ARN of your batch processing role. Our file looks like this:
+
+::
+
+    {
+      "jobDefinitionName": "merge-job",
+      "type": "container",
+      "containerProperties": {
+        "jobRoleArn": "arn:aws:iam::422553907334:role/spacetx-batch",
+        "image": "spacetx/merge-batch-job",
+        "vcpus": 1,
+        "memory": 2500
+      }
+    }
+
+Then from the directory where this file lives run the following command:
+
+``aws batch submit-job --cli-input-json file://register-merge-job.json``
+
+Again, check that your job has been successfully registered from the job console, our two jobs are ready to go!
+
+.. figure:: images/aws-tutorial-figure4.png
+   :align: center
+
+
+Run your Batch Jobs
+-------------------
+
+Now that we've set everything up it's time to run our jobs! The script `starfish-workflow.py` will handle submitting the process-fov array job
+then the merge job with a dependency on the first job to finish. All you'll need to do is run the script with a few parameters:
+
+::
+
+    --experiment-url: The url to your experiment.json file. Our is "https://s3.amazonaws.com/aws-processing-example/iss-spacetx-formatted/experiment.json"
+
+    --num-fovs: The number of fields of view in the experiment. We have 15
+
+    --recipe-location: The path to your recipe file in s3. Ours is "s3://aws-processing-example/aws-processing-example/iss-spacetx-formatted/recipe.py"
+
+    --results-location: The s3 bucket to copy the results from the job to. Ours is "s3://aws-processing-example/iss-spacetx-formatted-results/"
+
+    --job-queue: The name of your job queue to run your jobs. Ours is "first-run-job-queue"
+
+
+Now we run our script:
+
+::
+
+    $ python3 starfish-workflow.py \
+    >     --experiment-url "https://s3.amazonaws.com/aws-processing-example/iss-spacetx-formatted/experiment.json" \
```

Comment:
This accesses s3 through http.  Does the bucket policy even apply in this case?

Reply:
no but it does for copying the recipe file and then the results back to a bucket. Which is all in the same job

Reply:
Doesn't that imply the bucket needs to have public access for the HTTP endpoint to work?  I think it might be better to make this work over s3://.  I suspect it might just work.

Reply:
Do you mean use Experiment.from_json() with an s3 path? How would that work without passing credentials?

Reply:
If it's run from an EC2 instance with an IAM role, it should just work.

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials

`the AWS SDKs, AWS CLI, and Tools for Windows PowerShell automatically get the credentials from the EC2 instance metadata service and use them`

Reply:
You're right it works!

Reply:
Can you validate that this works on a bucket without public read access? I believe it should but it would be good to.